test/aleph/http_test.clj

```diff
@@ -570,3 +570,40 @@
                                       (.write ctx msg p)))))})
       (let [resp @(http-get (str "http://localhost:" port "/string"))]
         (is (= test-header-val (get (:headers resp) test-header-name)))))))
+
+(defn add-http-object-aggregator [^ChannelPipeline pipeline]
+  (.addBefore pipeline
+              "request-handler"
+              "http-object-aggregator"
+              (HttpObjectAggregator. 5)))
+
+(deftest test-http-object-aggregator-support
+  (with-server (http/start-server
+                basic-handler
+                {:port port
+                 :pipeline-transform add-http-object-aggregator})
+    (let [rsp @(http-put (str "http://localhost:" port "/echo")
+                         {:body "hello"})]
+      (is (= "hello" (bs/to-string (:body rsp))))
+      (is (= 200 (:status rsp))))
+
+    (let [rsp @(http-put (str "http://localhost:" port "/echo")
+                         {:body "hello, world!"})]
+      (is (= 413 (:status rsp)))
+      (is (empty? (bs/to-string (:body rsp)))))))
+
+(deftest test-http-object-aggregator-raw-stream-support
+  (with-server (http/start-server
+                basic-handler
```

Comment:
After a second pass on this PR, I would consider adding a `.release` on
the handler regarding `raw-stream?` to validate this :

> ;; Retain the content of FullHttpRequest one extra time 
;; compensate for it being released together with the request.

Reply:
Oh yeah, this is a bit sublte: For a while I was stumped by the fact that I couldn't trigger the leak detector because it seemed like I was leaking the buffer here. After digging into Netty a bit more, I learned why: Note how `echo-handler`  returns the `(:body request)` manifold stream verbatim as the response body. This leads to Netty itself releasing the buffers after writing them out to the wire. Makes sense when you think about it! Unfortunately, I was unable to find a reference document mentioning this but if it's anything to go by, the official user guide's ["Writing an Echo Server" section](https://netty.io/wiki/user-guide-for-4.x.html#wiki-h3-8) (which essentially does the same as we do here) mentions this in passing:

> It is because Netty releases it for you when it is written out to the wire.

So in fact, adding a `.release` call here would result in a double-free :grimacing: If you have an idea how to make this test rely on a little less context, I'm all ears, though :smile:

Reply:
TIL, thank you.